apps/backend/src/groups/groups.service.ts

```diff
@@ -142,4 +171,42 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  // This method ensures that the passed in user is in all of the
+  // passed in groups, as long as the group already exists.
+  // It will additionally remove the user from any groups not in the list.
+  // Called from oidc.strategy.ts, if OIDC_EXTERNAL_GROUPS is enabled
+  async syncUserGroups(user: User, groups: string[]) {
```

Comment:
at the end of this function, we should run the 'ensuregrouphasowner' function against any group that has been impacted by changes / potentially just against all groups entirely